### PR TITLE
Handled interests step during adherent registration

### PR DESCRIPTION
**Pull request by HeahDude**
Closes [EM-14](http://jira.sensiolabs.net/browse/EM-14).

### CODE REVIEW

File: src/AppBundle/Controller/MembershipController.php [hunk 1]
```diff
@@ -2,21 +2,28 @@
...
-use AppBundle\Form\MembershipRequestType;
-use Sensio\Bundle\FrameworkExtraBundle\Configuration\Entity;
-use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
-use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
-use Symfony\Component\Form\Extension\Core\Type\SubmitType;
-use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\Response;
+use AppBundle\{
+    Entity\Adherent,
```
Comment (hhamon) [inline src/AppBundle/Controller/MembershipController.php]: What?

File: src/AppBundle/Controller/MembershipController.php [hunk 2]
```diff
@@ -2,21 +2,28 @@
...
+    Entity\AdherentActivationToken,
+    Exception\AdherentAlreadyEnabledException,
+    Exception\AdherentTokenExpiredException,
+    Form\AdherentInterestsFormType,
+    Form\DonationType,
+    Form\MembershipRequestType,
+    Intl\UnitedNationsBundle,
+    Membership\MembershipRequest
+};
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\{
```
Comment (hhamon) [inline src/AppBundle/Controller/MembershipController.php]: What What?

File: src/AppBundle/Controller/MembershipController.php [hunk 3]
```diff
@@ -2,21 +2,28 @@
...
+    Entity,
+    Method,
+    Route
+};
+use Symfony\{
```
Comment (hhamon) [inline src/AppBundle/Controller/MembershipController.php]: What What What?

File: tests/AppBundle/Controller/MembershipControllerTest.php [hunk 4]
```diff
@@ -211,6 +211,85 @@ public function testDonateWithoutTemporaryDonation()
         $this->assertResponseStatusCode(Response::HTTP_NOT_FOUND, $client->getResponse());
     }
 
+    public function testPinInterestsWithoutNewAdherentId()
+    {
+        $client = $this->client;
```
Comment (hhamon) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: why?

File: tests/AppBundle/Controller/MembershipControllerTest.php [hunk 5]
```diff
@@ -211,6 +211,85 @@ public function testDonateWithoutTemporaryDonation()
...
+        $client->request(Request::METHOD_GET, '/inscription/centre-interets');
+
+        $this->assertResponseStatusCode(Response::HTTP_NOT_FOUND, $client->getResponse());
+    }
+
+    public function testPinInterestsWithWrongNewAdherentId()
+    {
+        $client = $this->client;
```
Comment (hhamon) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: why?

File: tests/AppBundle/Controller/MembershipControllerTest.php [hunk 6]
```diff
@@ -211,6 +211,85 @@ public function testDonateWithoutTemporaryDonation()
...
+        $client->getContainer()->get('session')->set(MembershipUtils::NEW_ADHERENT_ID, 'wrong id');
+        $client->request(Request::METHOD_GET, '/inscription/centre-interets');
+
+        $this->assertResponseStatusCode(Response::HTTP_NOT_FOUND, $client->getResponse());
+    }
+
+    public function testPinInterests()
+    {
+        $client = $this->client;
```
Comment (hhamon) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: why?

File: tests/AppBundle/Controller/MembershipControllerTest.php [hunk 7]
```diff
@@ -211,6 +211,85 @@ public function testDonateWithoutTemporaryDonation()
...
+
+        foreach ($checkboxes as $i => $checkbox) {
+            $this->assertSame($interestsValues[$i], $checkbox->getAttribute('value'));
+            $this->assertSame($interestsLabels[$i], $crawler->filter('label[for="app_adherent_pin_interests_interests_'.$i.'"]')->eq(0)->text());
+        }
+    }
+
+    public function testPinInterestsPersistsInterestsForNonActivatedAdherent()
+    {
+        $client = $this->client;
```
Comment (hhamon) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: why?

File: src/AppBundle/Controller/MembershipController.php (hunk 3 above)
Comment (HeahDude) [inline src/AppBundle/Controller/MembershipController.php]: Welcome to PHP 7 :D

Comment (hhamon) [inline src/AppBundle/Controller/MembershipController.php]: Mé saŸ mochE ^^

File: src/AppBundle/Controller/MembershipController.php [hunk 8]
```diff
@@ -97,6 +106,54 @@ public function donateAction(Request $request): Response
...
     /**
+     * This action enables a new user to pin his/here interests as a third step
+     * of the registration process, thus he/she is not logged-in/activated yet.
+     *
+     * @Route("/inscription/centre-interets", name="app_membership_pin_interests")
+     * @Method("GET|POST")
+     */
+    public function pinInterestsAction(Request $request): Response
+    {
+        $memberShipUtils = $this->get('app.membership_utils');
```
Comment (tgalopin) [inline src/AppBundle/Controller/MembershipController.php]: `membership`

File: tests/AppBundle/Controller/MembershipControllerTest.php (hunk 4 above)
Comment (HeahDude) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: Fixed.

File: tests/AppBundle/Controller/MembershipControllerTest.php (hunk 5 above)
Comment (HeahDude) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: Fixed.

File: tests/AppBundle/Controller/MembershipControllerTest.php (hunk 6 above)
Comment (HeahDude) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: Fixed.

File: tests/AppBundle/Controller/MembershipControllerTest.php (hunk 7 above)
Comment (HeahDude) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: Fixed.

File: src/AppBundle/Controller/MembershipController.php (hunk 8 above)
Comment (HeahDude) [inline src/AppBundle/Controller/MembershipController.php]: Fixed.

File: src/AppBundle/Form/AdherentInterestsFormType.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+use AppBundle\Entity\Adherent;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
```
Comment (hhamon) [inline src/AppBundle/Form/AdherentInterestsFormType.php]: Should be `/*`

File: tests/AppBundle/Controller/MembershipControllerTest.php [hunk 10]
```diff
@@ -214,6 +214,80 @@ public function testDonateWithoutTemporaryDonation()
...
+        $this->client->submit($crawler->selectButton('app_adherent_pin_interests[submit]')->form(), [
+            'app_adherent_pin_interests' => [
+                'interests' => $chosenInterests,
+            ],
+        ]);
+
+        $this->assertClientIsRedirectedTo('/inscription/choisir-des-comites', $this->client);
+
+        /* @var Adherent $adherent */
+        $adherent = $this->getAdherentRepository()->find(1);
```
Comment (hhamon) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: You must not hardcode an ID

Comment (hhamon) [inline tests/AppBundle/Controller/MembershipControllerTest.php]: Better to find by email address or UUID if known.

**Pull request merged**